### PR TITLE
WIP: update headlessui from v1 to v2

### PR DESCRIPTION
**Pull request by jeremiah-clothier**
### Test Plan:

Updating HeadlessUI from v1 to v2

* Release Notes: https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v2.0.0

- [ ] Wrote/updated [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [ ] CI tests / new tests are not applicable
- [ ] Manually tested my changes, and here are the details:

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -329,15 +329,16 @@ export const Modal = (props: ModalProps) => {
...
             styles['modal__overlay'],
             overlayEmphasis &&
               styles[`modal__overlay--emphasis-${overlayEmphasis}`],
           )}
         />
-
-        <ModalContent onClose={onClose} {...rest} />
+        <Dialog.Panel>
+          <ModalContent onClose={onClose} {...rest} />
+        </Dialog.Panel>
```
Comment (jeremiah-clothier) [inline src/components/Modal/Modal.tsx:341]: This is the only typescript error I got from the migration. Dialog.Overlay was deprecated in v1.6

https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v1.6.0

File: src/components/ToastNotification/ToastNotification.stories.tsx [hunk 2]
```diff
@@ -100,6 +100,7 @@ const ToastNotificationManager = (args: Args) => {
         {toasts.map((toast) => (
           <Transition
             appear
+            as="div"
```
Comment (jeremiah-clothier) [inline src/components/ToastNotification/ToastNotification.stories.tsx:103]: The default `as` for this component is now `as={React.Fragment}`

**Comment by booc0mtaco** [comment]
With headlessUI v2 we can also remove a few other dependencies that v1 needed (like popperJS and tippyJS? maybe just one of those or both... cant recall) But great to see it didn't cause much meaningful jitter; the goal of this repo is to abstract these internals :)

**Comment by jeremiah-clothier** [comment]
> With headlessUI v2 we can also remove a few other dependencies that v1 needed (like popperJS and tippyJS? maybe just one of those or both... cant recall) But great to see it didn't cause much meaningful jitter; the goal of this repo is to abstract these internals :)

+1 headlessUI changed all their imports from like `Dialog.Panel` to `DialogPanel` but it's only a deprecation so could be a separate PR after upgrading.

For posterity, we can also get rid of `ExtractProps` in EDS because headless ui v2 exports the props of each of their components (finally)

@booc0mtaco

**Pull request closed without merging**